### ISSUE TITLE
Where clause doesn't work with bool when record is migrated until the record is updated

### ISSUE THREAD

**Issue by VNGames** [praeclarum/sqlite-net#1245]
I have a table

```
public class FocusSelection
{
    [PrimaryKey, AutoIncrement]
    public int Id { get; set; }
    // some properties
}
```
I add a record into that table. It is there. Everything is fine. Now to implement a new feature I add the SomeProperty below.

Then update the app version and run it.
```
public class FocusSelection
{
    [PrimaryKey, AutoIncrement]
    public int Id { get; set; }
    // some properties
    public bool SomeProperty { get; set; }
}
```

At the very start of the app I execute the code.

```
await _databaseService.Connection.RunInTransactionAsync(connection => 
{
     // this returns that record with SomeProperty = false as expected
     var all = connection.Table<Models.Database.FocusSelection>().ToList();
     // But this returns no 0 records
     var t = connection.Table<Models.Database.FocusSelection>().Where(x => !x.SomeProperty).ToList();
}
```

And the only way to make query work is to do this

```
await _databaseService.Connection.RunInTransactionAsync(connection => 
{
     var all = connection.Table<Models.Database.FocusSelection>().ToList();
     // updating records
     all.ForEach(x => connection.Update(x));

     // Now this query works fine
     var t = connection.Table<Models.Database.FocusSelection>().Where(x => !x.SomeProperty).ToList();
}
```

**Comment by VNGames** [comment]
@Haukinger I think the reason for such behavior is this from the documentation
https://github.com/praeclarum/sqlite-net/wiki/Automatic-Migrations

`The automatic migration, currently, only supports adding new columns. If your classes have new properties that are not associated with columns in the table, then alter table ... add column commands will be executed to bring the database up to date. These new columns will not have default values and will therefore be null`

Pretty much even when a property is not nullable it sets NULL and then all queries can't match the actual value no matter what value that is (if it is a bool then true or false) because it is NULL. (Just tested this by expecting the database after migration using external viewer and it is indeed what's happening)

But if you update the record it gets a normal value and queries start to work.

**Comment by Haukinger** [comment]
That is, if the new column's `bool?`, it should work without the update, because `null`'s a valid value then?

**Comment by VNGames** [comment]
@Haukinger yes, it will work (the query will have to look like `!(x.Property ?? false)` to match `false` - haven't tested. Maybe it needs full `x.Property == null || !x.Property`). But it is dirty in case you don't really need `bool?` and using it just as a workaround. Updating all records is also dirty but if you have some system of manual migrations you can add this migration starting from some app version and after half a year when all users are using the app past this version you can remove this dirt (but first forcing old app versions to migrate over time, not a very reliable solution because you never know how much time somebody isn't using the app, and until all are migrated you can't remove the migration to update all records)

Another option is to try EF Core instead of this library. It is much more powerful and it can even have proper migrations but there are some hoops to jump over to setup it. Not critical though. I haven't used it but considering somewhere into the future since EF exists for a long time and it probably works well at least for basic stuff unlike this library. It also has proper entity relations by foreign key and queries using them.

**Comment by VNGames** [comment]
Also strange that even query like x => x.Property != 0 or x => x.Property != true also do not work even though != 0 should be matching NULL or any other number and Property != true should match NULL and false.
